### PR TITLE
sci-misc/boinc: Cleanup and upgraded init script

### PR DESCRIPTION
**Pull request by Yamakuzure**
- New ebuild boinc-7.2.42.ebuild
- Removed obsolete boinc-7.2.0.ebuild
- Removed obsolete boinc-7.2.0-r1.ebuild
- New ebuild boinc-7.4.42-r2.ebuild
- Removed obsolete boinc-7.4.42-r1.ebuild
- New ebuild boinc-7.6.31-r1.ebuild
- Removed obsolete boinc-7.6.31.ebuild
- Added VSYSCALL mapping check to all ebuilds
- Upgraded init script

Gentoo-Bug: 574260, 577490, 578750

Package-Manager: portage-2.2.28

@jlec @SoapGentoo I hope I did it right
@williamh Hopefully the init script is now up to scratch!

### CODE REVIEW

File: sci-misc/boinc/files/boinc.init [hunk 1]
```diff
@@ -22,24 +24,54 @@ create_work_directory() {
...
-		eerror "This script cannot be used for baselayout-1."
+
+env_check() {
+	# Make sure the configuration is sane
+	[ -z "$USER" ]             && USER="boinc"
+	[ -z "$GROUP" ]            && GROUP="boinc"
+	[ -z "$RUNTIMEDIR" ]       && RUNTIMEDIR="/var/lib/boinc"
+	[ -z "$BOINCBIN" ]         && BOINCBIN="$(which boinc_client)"
+	[ -z "$ALLOW_REMOTE_RPC" ] && ALLOW_REMOTE_RPC="yes"
+	[ -z "$NICELEVEL" ]        && NICELEVEL="19"
```
Comment (rindeal) [inline sci-misc/boinc/files/boinc.init]: this section can be simplified by using `: ${VAR:="DEFAULT_VALUE"}` syntax

File: sci-misc/boinc/files/boinc.init [hunk 2]
```diff
@@ -50,24 +82,39 @@ start() {
...
 		ewarn "Note that for attaching to some project you need your network up and running."
 		ewarn "network is needed only for jobs fetching afterwards"
 	fi
 
+	return 0
+}
+
+
+start() {
+	start_pre || return 1
```
Comment (rindeal) [inline sci-misc/boinc/files/boinc.init]: `*_(pre|post)` are run automatically

**Comment by SoapGentoo** [comment]
@Yamakuzure could you add the suggested improvements? Then I can merge it

**Comment by Yamakuzure** [comment]
@rindeal  @SoapGentoo : I have added the suggested improvements. Thank you very much for your feedback, I am learning a lot! :-)

**Comment by SoapGentoo** [comment]
@Yamakuzure Thanks. Could you please minimize the number of commits? i.e. you added a new ebuild, and then later on fixed some things. Use `git -rebase -i` (http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) to reorder and squash commits so we can have a nice and clear history.

**Comment by Yamakuzure** [comment]
@SoapGentoo : I have squashed the 10 commits into 8. One for each ebuild addition/removal and one for the init script update. Thanks to the 1-commit-per-file policy, re-ordering the commits to be able to squash them was simple. :-)

**Comment by SoapGentoo** [comment]
@Yamakuzure, thanks for your commitment, great work! **ONE** (😜) more tiny thing: could you remove the ".ebuild" from the COMMITMSGs, as it serves no purpose and is not Gentoo-idiomatic.

**Comment by Yamakuzure** [comment]
@SoapGentoo I hope this is correct now? (Awesome what 'git rebase -i' can do!)

**Comment by SoapGentoo** [comment]
@Yamakuzure, indeed, git rebase can do wonders. One last thing (that I should have said in my last comment, sorry): Never add full-stops at the end of the first line. Could you remove those and then I promise to shut up 😄.

**Comment by Yamakuzure** [comment]
@SoapGentoo : Done. (*pantpantpant*)

**Comment by SoapGentoo** [comment]
@Yamakuzure, please also add commas after "ebuild"... just kidding. Thanks for the great work, merged.

**Comment by thican** [comment]
Sorry guys, but the function env_check is horribly wrong. Hopely for me I caught this nasty before rebooting.

You used the parameter expansion "${PARAMETER:=WORD}" for assignment, but the problem is you are doing it on a line as is, therefore its content is executed, while it obviously shouldn't.

I don't know how to use correctly github, and no time to try, but here some of the modifications (I didn't check everything):
 env_check() {
 	# Make sure the configuration is sane
-	${USER:="boinc"}
-	${GROUP:="boinc"}
-	${RUNTIMEDIR:="/var/lib/boinc"}
-	${BOINCBIN:="$(which boinc_client)"}
-	${ALLOW_REMOTE_RPC:="yes"}
-	${NICELEVEL:="19"}
+	USER=${USER:-"boinc"}
+	GROUP=${GROUP:-"boinc"}
+	RUNTIMEDIR=${RUNTIMEDIR:-"/var/lib/boinc"}
+	BOINCBIN=${BOINCBIN:-"$(which boinc_client)"}
+	ALLOW_REMOTE_RPC=${ALLOW_REMOTE_RPC:-"yes"}
+	NICELEVEL=${NICELEVEL:-"19"}
 	# ARGS is not checked, it could have been explicitly set
 	# to be empty by the user.

Therefore, I wonder if you even tried once your code…

Anyway, have a nice day.

**Comment by SoapGentoo** [comment]
Indeed, setting defaults should be done using `VAR="${VAR:-default value}"` (like in your example). @Yamakuzure, could you send in a new PR with the fixes? Thanks.

**Comment by rindeal** [comment]
There is no need for the duplication of var names by using `USER=${USER:-"DEFAULT}"` syntax. Simply prepend `: ` as suggested in [my original comment](https://github.com/gentoo/gentoo/pull/1374#discussion_r61568145) and it should be fine.

**Comment by thican** [comment]
Indeed, `: ${USER:="boinc"} is a nice solution, didn't know about this feature.

(On your comment, the background is #f5f5f5 on pure white, didn't notice the ":" was part of it)

**Comment by SoapGentoo** [comment]
Fixed.

**Comment by rindeal** [comment]
ref https://github.com/gentoo/gentoo/commit/790b10547476baba875701da659349976cadd0ff

**Comment by Yamakuzure** [comment]
@rindeal I am Sorry, I didn't realize the colon was part of the statement.

**Pull request closed without merging**